common/src/IO/CharArrayReader.cpp

```diff
@@ -21,6 +21,7 @@
 
 #include <cassert>
 #include <cstring>
+#include <functional>
```

Comment:
std::less_equal is used in this file and it is defined in <functional>
http://en.cppreference.com/w/cpp/utility/functional/less_equal

Otherwise it won't compile on MSVC 2015